core/templates/dev/head/components/summary_tile/ExplorationSummaryTileDirective.js

```diff
@@ -130,20 +130,29 @@ oppia.directive('explorationSummaryTile', [
               return '#';
             } else {
               var result = '/explore/' + $scope.getExplorationId();
-              if ($scope.getCollectionId()) {
+              var urlParams = UrlService.getUrlParams();
+              var parentExplorationIds = $scope.getParentExplorationIds();
+
+              var collectionIdToAdd = $scope.getCollectionId();
+              // Replace the collection ID with the one in the URL if it exists
+              // in urlParams.
+              if (parentExplorationIds &&
+                  urlParams.hasOwnProperty('collection_id')) {
+                collectionIdToAdd = urlParams.collection_id;
```

Comment:
Isn't `urlParams.collection_id` same as `$scope.getCollectionId()` ?

Reply:
No, it isn't!

The refresher exploration might outside the collection that the parent exploration belongs to. This was what was causing the bug.

Reply:
Oh, you mean a redirection from a refresher exploration having a collection_id in url will go to another refresher exploration without preserving collection_id?

Reply:
Yes, that's right!

Reply:
Ok, thanks! LGTM